api/http/handler/edgestacks/edgestack_status_update.go

```diff
@@ -95,16 +97,16 @@ func (handler *Handler) updateEdgeStackStatus(tx dataservices.DataStoreTx, r *ht
 			return nil, nil
 		}
 
-		return nil, err
+		return nil, utils.NewEdgeError(strconv.Itoa(int(payload.EndpointID)), err)
 	}
 
 	endpoint, err := tx.Endpoint().Endpoint(payload.EndpointID)
 	if err != nil {
-		return nil, handler.handlerDBErr(err, "Unable to find an environment with the specified identifier inside the database")
+		return nil, handler.handlerDBErr(utils.NewEdgeError(strconv.Itoa(int(payload.EndpointID)), err), "Unable to find an environment with the specified identifier inside the database")
 	}
 
 	if err := handler.requestBouncer.AuthorizedEdgeEndpointOperation(r, endpoint); err != nil {
-		return nil, httperror.Forbidden("Permission denied to access environment", err)
+		return nil, httperror.Forbidden("Permission denied to access environment", utils.NewEdgeError(endpoint.Name, err))
```

Comment:
```suggestion
		return nil, httperror.Forbidden("Permission denied to access environment", fmt.Errorf("unauthorized edge endpoint operation: %w. Environment name: %s", err, endpoint.Name))
```

Reply:
Using this pattern is also flexible, we can tailor the message with the Environment name here VS Environment ID above.